pydantic_ai_slim/pydantic_ai/models/openai.py

```diff
@@ -672,6 +706,52 @@ def _get_web_search_options(self, model_request_parameters: ModelRequestParamete
                     f'`{tool.__class__.__name__}` is not supported by `OpenAIChatModel`. If it should be, please file an issue.'
                 )
 
+    def _map_model_response(self, message: ModelResponse) -> chat.ChatCompletionMessageParam:
+        """Hook that determines how `ModelResponse` is mapped into `ChatCompletionMessageParam` objects before sending.
+
+        Subclasses of `OpenAIChatModel` may override this method to provide their own mapping logic.
+        """
+        texts: list[str] = []
+        tool_calls: list[ChatCompletionMessageFunctionToolCallParam] = []
+        for item in message.parts:
+            if isinstance(item, TextPart):
+                texts.append(item.content)
+            elif isinstance(item, ThinkingPart):
+                # NOTE: DeepSeek `reasoning_content` field should NOT be sent back per https://api-docs.deepseek.com/guides/reasoning_model,
+                # but we currently just send it in `<think>` tags anyway as we don't want DeepSeek-specific checks here.
+                # If you need this changed, please file an issue.
+                start_tag, end_tag = self.profile.thinking_tags
+                texts.append('\n'.join([start_tag, item.content, end_tag]))
+            elif isinstance(item, ToolCallPart):
+                tool_calls.append(self._map_tool_call(item))
+            # OpenAI doesn't return built-in tool calls
+            elif isinstance(item, BuiltinToolCallPart | BuiltinToolReturnPart):  # pragma: no cover
+                pass
+            elif isinstance(item, FilePart):  # pragma: no cover
+                # Files generated by models are not sent back to models that don't themselves generate files.
+                pass
+            else:
+                assert_never(item)
+        message_param = chat.ChatCompletionAssistantMessageParam(role='assistant')
+        if texts:
+            # Note: model responses from this model should only have one text item, so the following
+            # shouldn't merge multiple texts into one unless you switch models between runs:
+            message_param['content'] = '\n\n'.join(texts)
+        else:
+            message_param['content'] = None
+        if tool_calls:
+            message_param['tool_calls'] = tool_calls
+        return message_param
+
+    def _map_finish_reason(
+        self, key: Literal['stop', 'length', 'tool_calls', 'content_filter', 'function_call']
+    ) -> FinishReason | None:
+        """Hooks that maps a finish reason key to a [FinishReason](pydantic_ai.messages.FinishReason).
```

Comment:
Linking to API docs by identifier requires square brackets; please check if we need this change anywhere else

```suggestion
        """Hooks that maps a finish reason key to a [FinishReason][pydantic_ai.messages.FinishReason].
```